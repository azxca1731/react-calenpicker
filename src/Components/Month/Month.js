import React from "react";
import PropTypes from "prop-types";
import { CssConnector, PropsConnector } from "Containers/Provider";
import dark from "./Month.style.dark";
import light from "./Month.style.light";

class Month extends React.Component {
  constructor(props) {
    super(props);
    this.style = props.theme == "light" ? light : dark;
  }

  componentDidMount() {}

  render() {
    const { month, cssObject } = this.props;
    return (
      <div className={this.style.Month} style={cssObject}>
        {month}
      </div>
    );
  }
}

Month.defaultProps = {};

Month.propTypes = {
  month: PropTypes.string.isRequired,
  cssObject: PropTypes.object,
  theme: PropTypes.string
};

export default PropsConnector(({ state }) => ({
  theme: state.theme
}))(
  CssConnector(({ state }) => ({
    cssObject: state.MonthCssObject
  }))(Month)
);
