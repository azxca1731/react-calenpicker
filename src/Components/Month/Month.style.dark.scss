@import "../../Styles/theme/dark";
@import "./Month.style";
