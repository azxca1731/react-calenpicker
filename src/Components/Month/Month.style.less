@import "../../Styles/colors.less";
.Month--dark {
  color: white;
  font-size: 30px;
}

.Month {
  color: @font-color;
  font-size: 30px;
}