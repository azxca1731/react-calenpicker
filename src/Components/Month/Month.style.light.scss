@import "Styles/theme/light";
@import "./Month.style";