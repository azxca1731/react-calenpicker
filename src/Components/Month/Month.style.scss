.Month {
  color: $font-color;
  font-size: 150%;
}
