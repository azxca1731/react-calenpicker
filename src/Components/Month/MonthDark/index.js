import MonthDark from "./MonthDark";

export default MonthDark;
