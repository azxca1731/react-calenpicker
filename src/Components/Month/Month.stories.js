import React from "react";

import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";

import Month from "Components/Month";

storiesOf("Components", module)
  .addDecorator(withInfo)
  .add("Month", () => <Month />);
