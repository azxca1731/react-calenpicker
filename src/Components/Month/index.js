import Month from "./Month";

export default Month;
