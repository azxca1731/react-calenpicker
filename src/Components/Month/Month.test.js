import React from "react";
import { shallow } from "enzyme";
import Month from "index";

describe("Month", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<Month />);
    expect(wrapper).toHaveLength(1);
  });
});
