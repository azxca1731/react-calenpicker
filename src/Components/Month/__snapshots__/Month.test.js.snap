// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month render custom css render well 1`] = `
<styled.div
  style={
    Object {
      "color": "red",
    }
  }
>
  2018-10
</styled.div>
`;

exports[`Month render with snapshot 1`] = `
<styled.div
  style={Object {}}
>
  2018.02
</styled.div>
`;
